cmd/system-probe/api/server.go

```diff
@@ -74,7 +82,7 @@ func StartServer(cfg *config.Config, telemetry telemetry.Component) error {
 
 	go func() {
 		err = srv.Serve(conn.GetListener())
-		if err != nil && err != http.ErrServerClosed {
+		if errors.Is(err, http.ErrServerClosed) {
```

Comment:
This should be `err != nil && !errors.Is(...)`?

Reply:
Yup, good catch.